starknet_py/net/signer/stark_curve_signer.py

```diff
@@ -69,11 +69,13 @@ def _sign_transaction(self, transaction: Invoke):
             contract_address=self.address,
             entry_point_selector=0
             if transaction.version == 1
-            else transaction.entry_point_selector,
+            else cast(int, transaction.entry_point_selector),
             calldata=transaction.calldata,
             max_fee=transaction.max_fee,
             chain_id=self.chain_id.value,
-            additional_data=[transaction.nonce] if transaction.version == 1 else [],
+            additional_data=[cast(int, transaction.nonce)]
+            if transaction.version == 1
+            else [],
```

Comment:
This would be merged around starknet 0.11.0, in which transactions in version 0 are removed.
I think we should wait till then and not support txs v0 in our dataclasses.